data_structures/src/transaction_factory.rs

```diff
@@ -5,57 +5,70 @@ use crate::{
     },
     error::TransactionError,
     transaction::{DRTransactionBody, VTTransactionBody},
-    utxo_pool::{OwnUnspentOutputsPool, UnspentOutputsPool, UtxoDiff, UtxoSelectionStrategy},
+    utxo_pool::{
+        NodeUtxos, OwnUnspentOutputsPool, UnspentOutputsPool, UtxoDiff, UtxoSelectionStrategy,
+    },
 };
 use std::{collections::HashSet, convert::TryFrom};
 
+/// Abstraction used to create transactions using a UtxoSet
+pub trait UtxoFantasy {
+    fn sort_iter(&self, strategy: UtxoSelectionStrategy) -> Vec<OutputPointer>;
+    fn get_time_lock(&self, outptr: &OutputPointer) -> Option<u64>;
+    fn get_value(&self, outptr: &OutputPointer) -> Option<u64>;
+    fn get_included_block_number(&self, outptr: &OutputPointer) -> Option<Epoch>;
+    fn set_used_output_pointer(&mut self, outptr: &OutputPointer, ts: u64);
+}
+
 /// Select enough UTXOs to sum up to `amount`.
 ///
 /// On success, return a list of output pointers and their sum.
 /// On error, return the total sum of the output pointers in `own_utxos`.
-pub fn take_enough_utxos(
-    own_utxos: &mut OwnUnspentOutputsPool,
-    all_utxos: &UnspentOutputsPool,
+pub fn take_enough_utxos<T>(
```

Comment:
I think this can now be refactored into a [default implementation](https://doc.rust-lang.org/book/ch10-02-traits.html#default-implementations) in the `UtxoFantasy` trait, without any need for generics.

Reply:
Although maybe making it a separate thing has some advantage for testing or something else? IDK. Please @tmpolaczyk chime in.